gv.c

```diff
@@ -2600,9 +2610,27 @@ Perl_gv_fetchpvn_flags(pTHX_ const char *nambeg, STRLEN full_len, I32 flags,
     if (!stash && !find_default_stash(&stash, name, len, is_utf8, add, sv_type)) {
         return NULL;
     }
+    char tmp[30];
+    char *use_name = (char*)name;
```

Comment:
use_name could be `const char *` if you change the name parameter to S_is_internal_caret_name() to match, avoiding this cast.